### ISSUE TITLE
Enhancement: Edit Tool Data Table Entries

### ISSUE THREAD

**Issue by scholtalbers** [galaxyproject/galaxy#6819]
It would be great to be able to edit Tool Data Table Entries from the admin interface instead of having to find out which *.loc file to manually edit.

See also http://dev.list.galaxyproject.org/tool-data-confusion-td4671600.html

**Comment by mvdbeek** [comment]
Agreed. A very simple text editor that highlights tabs and does simple validation should do.
You can find a list of all usable tables in admin -> data_tables, that might help a little bit.

**Comment by bernt-matthias** [comment]
Two comments from my side: 

1. Automation of setting tool data table entries would be more interesting (for me), than manual editing. But I guess a simple web editor would come with corresponding API capabilities.. I guess. 

2. Note that the dev-list mail was mainly about the confusion caused by the two xml files:

- `config/shed_tool_data_table_conf.xml`
- 
`tool-data/toolshed.g2.bx.psu.edu/repos/REPO/TOOL/VERSION/tool_data_table_conf.xml`

and two loc files:

- `tool-data/TOOLDB.loc`
- `tool-data/toolshed.g2.bx.psu.edu/repos/REPO/TOOL/VESRION/TOOLDB.loc` 

where only the latter is refered to by both xml files.

**Comment by mvdbeek** [comment]
> * Automation of setting tool data table entries would be more interesting (for me), than manual editing. But I guess a simple web editor would come with corresponding API capabilities.. I guess.
> *

This is complicated, loc file entries are completely unstructured, and so they may contain anything. Paths, partial paths, prefixes, suffixes, URLs, you name it.

The loc files for the same data table get merged (in-memory) on startup, and when installing a completely new table we copy the correct sample files to `tool-data/`. So if you're going to manually edit stuff, I'd do that on the loc file in `tool-data/`

**Comment by bernt-matthias** [comment]
Thanks for the comment. 

I also tend to edit the `loc` files in `tool-data/`. What is still bugging me is the initial setup of the entries in the xml files which point redundantly to the same loc file within `tool-data/toolshed.g2.bx.psu.edu/...`. I'm still not sure if this is a feature or a bug... :)

Also, with the current state the default setup leads to the merging of 2 copies of the same file for each tool, or?

**Comment by FredericBGA** [comment]
Hi,
It could be nice to at least see the number of entries. 
Right now there is 

Name | Filename | Tool data path | Errors
-- | -- | -- | --

would be nice:

Name | Filename | Entries | Tool data path | Errors
-- | -- | -- | -- | --

And display this number close to the reload button as well:
`Data Manager: blastdb_p 150`
The manager can so easily see if the number has changed after hitting the reload button.
